Source/JavaScriptCore/parser/ASTBuilder.h

```diff
@@ -197,9 +197,9 @@ class ASTBuilder {
     bool isMetaProperty(ExpressionNode* node) { return node->isMetaProperty(); }
     bool isNewTarget(ExpressionNode* node) { return node->isNewTarget(); }
     bool isImportMeta(ExpressionNode* node) { return node->isImportMeta(); }
-    ExpressionNode* createResolve(const JSTokenLocation& location, const Identifier& ident, const JSTextPosition& start, const JSTextPosition& end)
+    ExpressionNode* createResolve(const JSTokenLocation& location, const Identifier& ident, const JSTextPosition& start, const JSTextPosition& end, const bool needToCheckUsesArguments = true)
```

Comment:
`needToCheckUsesArguments` should be `false` by default. We should mark `usesArguments` by default, and allowing to skip it only when we are focusing on `arguments.length`.

Reply:
`needToCheckUsesArguments` should be `true` by default since

```
    ExpressionNode* createResolve(const JSTokenLocation& location, const Identifier& ident, const JSTextPosition& start, const JSTextPosition& end, const bool needToCheckUsesArguments = true)
    {
        if (needToCheckUsesArguments && m_vm.propertyNames->arguments == ident)
            usesArguments();
        ...
```